File: baker-interface/src/main/scala/com/ing/baker/runtime/javadsl/EventResolutions.scala
package com.ing.baker.runtime.javadsl

import java.util.concurrent.CompletableFuture

import com.ing.baker.runtime.common
import com.ing.baker.runtime.common.LanguageDataStructures.JavaApi

case class EventResolutions(resolveWhenReceived: CompletableFuture[SensoryEventStatus],
                            resolveWhenCompleted: CompletableFuture[SensoryEventResult]
) extends com.ing.baker.runtime.common.EventResolutions[CompletableFuture] with JavaApi {

  type SensoryEventResultType = SensoryEventResult

  def getResolveWhenReceived: CompletableFuture[SensoryEventStatus] = resolveWhenReceived

  def getResolveWhenCompleted: CompletableFuture[SensoryEventResult] = resolveWhenCompleted
}

File: core/baker-interface-kotlin/src/main/kotlin/com/ing/baker/runtime/kotlindsl/EventResolutions.kt
package com.ing.baker.runtime.kotlindsl

import com.ing.baker.runtime.common.SensoryEventStatus
import com.ing.baker.runtime.javadsl.SensoryEventResult
import kotlinx.coroutines.Deferred

data class EventResolutions(
    val resolveWhenReceived: Deferred<SensoryEventStatus>,
    val resolveWhenCompleted: Deferred<SensoryEventResult>
)